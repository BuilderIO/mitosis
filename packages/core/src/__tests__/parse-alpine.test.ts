import { describe } from 'vitest';
import { runTestsForAlpineSyntax } from './test-generator';

describe('Parse Alpine', () => {
  runTestsForAlpineSyntax();
});
