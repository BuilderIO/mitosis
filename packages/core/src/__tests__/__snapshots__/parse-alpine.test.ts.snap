// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parse Alpine > basic 1`] = `
{
  "@type": "@builder.io/mitosis/component",
  "children": [
    {
      "@type": "@builder.io/mitosis/node",
      "bindings": {},
      "children": [
        {
          "@type": "@builder.io/mitosis/node",
          "bindings": {
            "data": {
              "bindingType": "expression",
              "code": "{ message: 'Hello Alpine!' }",
              "type": "single",
            },
          },
          "children": [
            {
              "@type": "@builder.io/mitosis/node",
              "bindings": {
                "_text": {
                  "bindingType": "expression",
                  "code": "\\"
  \\"",
                  "type": "single",
                },
              },
              "children": [],
              "meta": {},
              "name": "div",
              "properties": {},
              "scope": {},
            },
            {
              "@type": "@builder.io/mitosis/node",
              "bindings": {
                "_text": {
                  "bindingType": "expression",
                  "code": "message",
                  "type": "single",
                },
              },
              "children": [],
              "meta": {},
              "name": "h1",
              "properties": {},
              "scope": {},
            },
            {
              "@type": "@builder.io/mitosis/node",
              "bindings": {
                "_text": {
                  "bindingType": "expression",
                  "code": "\\"
\\"",
                  "type": "single",
                },
              },
              "children": [],
              "meta": {},
              "name": "div",
              "properties": {},
              "scope": {},
            },
          ],
          "meta": {},
          "name": "div",
          "properties": {},
          "scope": {},
        },
        {
          "@type": "@builder.io/mitosis/node",
          "bindings": {
            "_text": {
              "bindingType": "expression",
              "code": "\\" \\"",
              "type": "single",
            },
          },
          "children": [],
          "meta": {},
          "name": "div",
          "properties": {},
          "scope": {},
        },
      ],
      "meta": {},
      "name": "div",
      "properties": {},
      "scope": {},
    },
  ],
  "context": {
    "get": {},
    "set": {},
  },
  "hooks": {
    "onEvent": [],
    "onMount": [],
  },
  "imports": [],
  "inputs": [],
  "meta": {},
  "name": "AlpineComponent",
  "props": {},
  "refs": {},
  "state": {},
  "subComponents": [],
}
`;
